### DISCUSSION TITLE
Is this repo unmantained?

### DISCUSSION THREAD

**Discussion by mateusmlo** [diegomura/react-pdf#1152] [General]
There haven been no releases since 2019 and I'm looking to use this lib on a professional project, but I'd rather not use something abandoned. Is there a mantained fork or someone who's taking care of things around?

**Comment by Chagall** [comment]
I question myself too if this project is still maintained. However, I've been using this lib for the past few months at my job and so far it has been the best experience I've had. It works pretty well with my React projects. 
The downside is that it lacks examples, so sometimes you have to figure things out by yourself, but It's totally possible to build production quality PDFs with this.

**Comment by herrfugbaum** [comment]
https://github.com/diegomura/react-pdf/pull/617#issuecomment-760596576 says

> Hello everybody. Sorry for being unresponsive. For awhile now I'm struggling with finding time to release this version (which still has some TODOs) and work on the lib overall. I want so much to be able to focus and work on react-pdf but it's been simply to hard to do it with a regular job and some other personal stuff. Hope I find time soon!

@diegomura I wonder if there is any way the community can help this library going forward. Maybe by moderating the issue section and closing off duplicate issues to reduce noise (search for "React 17" for example)?

Also for everyone who wants to support this project: There is an open collective link where you can donate if you are able to. I assume this helps best since it supports dev time.

**Comment by diegomura** [comment]
Thank you all!

Answering the discussion's question: no, it's not 😄 

Can't deny it's been hard to keep track of everything since I'm mostly the only one coding, but that's mostly my fault for not being more communicative about what should be done. If someone has any good ideas about what I can do to make the project more welcoming to devs to contribute, please do! Flagging repeated issues and answering discussions and issues (such as many people do already!) is super helpful as well. Thanks for that.

The reason the master branch does not have may releases is because I've been working on a v2.0 version which is very closely to be merged as official. I think this has also mad it difficult for people to contribute since both solutions have nothing to do. I'm also a bit afraid that the way the new version is structured might make people not to contribute since it follows a more functional approach using rambda. I can explain the reasons behind this separately if someone's curious, but I'm also open to suggestions there.

**Reply by Chagall** [reply]
Since you asked about how to make it more welcoming, I ask: Is there a place where the project structure is explained, at least a bit, so people know how things work and where things are (in terms of what goes into each folder and so on)? I know the site explains the lifecycle of how the document is renderem, but at least for me I quite don't get yet how everything is structured.

**Reply by Chagall** [reply]
And, I'd like to know a bit more about this 2.0 version if you don't mind. Is the new version going to be incompatible with the old one? And will the old one still be available on npm after 2.0 gets released?

**Reply by diegomura** [reply]
First of all, thank you @Chagall for being so active on the project lately. Your help is super valuable!

>  Is there a place where the project structure is explained, at least a bit, so people know how things work and where things are?

Unfortunately not. This is a great point of improvement and I'll try to write some of this stuff down so people who wants to contribute knows how. However, 2.0 changes everything internally (some might argue that will make things a bit more complex, or at least out of the ordinary), so I'll probably write it about 2.0 right away.

> Is the new version going to be incompatible with the old one?

No! 2.0 will be backwards compatible. The reason why the major version will be increased it's mostly because 2.0 is a complete reimplementation of the library. The reasons about why I choose to do this are many. Most of them can be read in the 2.0 branch description here https://github.com/diegomura/react-pdf/pull/617 under `Goals behind this idea` and `Other cool ideas in mind`. That does not includes being easier to test (2.0 has more than 1k unit tests, testing almost every API and a vast number of use cases).

One of the remaining things to do so I can ship this version is writing a blog post explaining all this as well.

> Will the old one still be available on npm after 2.0 gets released?

Yes it will. I cannot delete it even if I want to, which I don't. I do not expect 2.0 transition to be so smooth (although it should), so the latest 1.X will be there as a fallback during this period, and also for anyone who wants to use it. However, it will probably be deprecated in terms of development. If it's just simple changes or fixes, I'll be able to do them for sure, but no major feature will be added on it. That mostly responds to a lack of time (being mostly myself who currently contributes, with suggestions such as yours that might change!), and desire to focus on the next version which I believe it's better and more stable.
